impl/src/main/java/slack/ThreadMessageHandler.java

```diff
@@ -0,0 +1,161 @@
+        AnswerResource answerResource
+    ) {
+        this.questionResource = questionResource;
+        this.slackResource = slackResource;
+        this.userResource = userResource;
+        this.answerResource = answerResource;
+
+    }
+
+    @Override
+    public boolean shouldHandle(JsonObject body) {
+        return body.has(SLACK_THREAD_ID) && !body.has("message") && !body.has("bot_id");
+    }
+
+    @Override
+    public Observable<Void> handleMessage(JsonObject message) {
+
+        return
+            questionResource
+                .getQuestionBySlackThreadId(message.get(SLACK_THREAD_ID).getAsString())
+
+                //No thread found -> create new
+                .onErrorResumeNext(throwable -> {
+                    if (throwable instanceof WebException) {
+                        if (NOT_FOUND.equals(((WebException)throwable).getStatus())) {
+                            return createQuestionAndPostToSlack(message);
+                        }
+                    }
+                    return error(throwable);
+                })
+                .flatMap(question -> createAnswer(message));
+    }
+
+    private Observable<Question> createQuestionAndPostToSlack(JsonObject message) {
+        return concat(createMainQuestion(message), postToSlack(message).cast(Question.class));
+    }
+
+    /**
+     * Posts reply to thread in slack
+     * @param message the thread
+     * @return
+     */
+    private Observable<Void> postToSlack(JsonObject message) {
+        return slackResource.postMessageToSlack(
+            message.get("channel").getAsString(),
+            "This looks like an interesting conversation, added it to slackoverflow",
+            message.get(SLACK_THREAD_ID).getAsString())
+        .ignoreElements();
+    }
+
+    /**
+     * Start the thread in our internal storage
+     * @param message message from slack
+     *
+     * @return
+     */
+    private Observable<Question> createMainQuestion(JsonObject message) {
+        String mainMessageId = message.get(SLACK_THREAD_ID).getAsString();
+        String channel       = message.get("channel").getAsString();
+
+        return
+            slackResource.getMessageFromSlack(channel, mainMessageId)
+                .flatMap(mainMessage -> slackResource.getUserEmail(mainMessage.getUser())
+                    .flatMap(email -> userResource.getUserByEmail(email, true))
+                    .map(User::getId)
+                    .flatMap(userId -> {
+                        Question question = new Question();
+
+                        question.setTitle(mainMessage.getText());
+                        question.setUserId(userId);
+                        question.setQuestion(mainMessage.getText());
+                        question.setSlackThreadId(mainMessageId);
+                        question.setBounty(DEFAULT_BOUNTY);
+
+                        return first(questionResource.postQuestion(getAuthAs(userId), question).doOnError(throwable -> {
+
```

Comment:
maybe log here? we might need to think about how expections should be propagated through the handlers in the future?